### ISSUE TITLE
Spatial fields are BLOB's but not hexed

### ISSUE THREAD

**Issue by basz** [ifsnop/mysqldump-php#103]
Apparently a long standing bug in MySQL but indeed this library also dumps POINT data as binary data and not as HEX. I just found out my backups are useless... :-/  Luckily no emergency...

from `man mysqldump`
> Dump binary columns using hexadecimal notation (for example, 'abc' becomes 0x616263). The affected data types are BINARY, VARBINARY, the BLOB types, and BIT.

I think your lib could easily add HEX support these for these column types?

GEOMETRY
POINT
LINESTRING
POLYGON
MULTIPOINT
MULTILINESTRING
MULTIPOLYGON
GEOMETRYCOLLECTION

**Comment by basz** [comment]
http://bugs.mysql.com/bug.php?id=43544

> Noted in 5.6.21, 5.7.5 changelogs.
> Internally, spatial data types such as Geometry are represented as
> BLOB values, so when invoked with the --hex-dump option, mysqldump
> now displays spatial values in hex.

**Comment by ifsnop** [comment]
Looks easy,  but before adding support I would need an example to add some
tests. Could you provide some table definition with all the column types
and some data.

Thanks in advance.

**Comment by basz** [comment]
After a bit more experimenting I found my POINT column had some incorrect binary data in it. That was the reason my backups couldn't be imported. When I simple nulled those incorrect records, the backup could be exported/imported just fine - even with the binary data not stored as hex.

That said HEX'ing geo data might still be a nice addition (mysqldump does it correctly now too) so I did prepare a simple schema for you to test with.

https://gist.github.com/basz/85dcbcc3a6d5b4be6add9ca47a46b4d8

It contains INSERTS from text and inserts as they came out of phpmyadmin - shown in binary data. The data for each column should match. I'm running MySql Community Server 5.7.12.

Thanks
